### ISSUE TITLE
`const`ify some of the methods

### ISSUE THREAD

**Issue by Kyllingene** [Auburn/FastNoiseLite#137]
I'm currently writing a voxel terrain generator (shush), and naturally I'm using perlin noise for that. Your crate has a very nice API, and thanks to its simplicity, converting most of the methods into `const` methods should be fairly straightforward.

I'd be glad to PR this myself, if you'd like.

**Comment by Auburn** [comment]
Sure, if you want to PR that would be great

**Comment by Keavon** [comment]
Please do submit a PR for this! I'll review it when it's ready, and we can publish a new release after the just-merged #138 and your PR are both landed.

**Comment by Kyllingene** [comment]
I'll do that this evening. One thing to note is that while this isn't a breaking change, reverting it will be; I'll take care to mention any possible pain points in my PR.

**Comment by Keavon** [comment]
Yep, as long as we're only adding `const` (which there shouldn't be any need to revert), this can be a point release.

@Auburn I think the best way to handle versioning when faced with the many different languages is that you plan to keep the the `x.y` part of semver's `x.y.z` format in sync across all ports, but the `.z` part can freely get out of sync between ports. Maybe the readme can mention that in the part that lists each language.
